notebooks/custom-prompts-demo.ipynb

```diff
@@ -0,0 +1,571 @@
+    "        self.customer_id = customer_id\n",
+    "        self.corpus_id = corpus_id\n",
+    "        self.api_key = api_key\n",
+    "        self.prompt_name = prompt_name if prompt_name else \"vectara-summary-ext-v1.2.0\"\n",
+    "        self.prompt_text = prompt_text\n",
+    "\n",
+    "    def get_body(self, query_str: str):\n",
+    "        corpora_key_list = [{\n",
+    "                'customer_id': self.customer_id, 'corpus_id': self.corpus_id, 'lexical_interpolation_config': {'lambda': 0.0}\n",
+    "            }\n",
+    "        ]\n",
+    "        body = {\n",
+    "            'query': [\n",
+    "                { \n",
+    "                    'query': query_str,\n",
+    "                    'start': 0,\n",
+    "                    'numResults': 50,\n",
+    "                    'corpusKey': corpora_key_list,\n",
+    "                    'context_config': {\n",
+    "                        'sentences_before': 2,\n",
+    "                        'sentences_after': 2,\n",
+    "                        'start_tag': \"%START_SNIPPET%\",\n",
+    "                        'end_tag': \"%END_SNIPPET%\",\n",
+    "                    },\n",
+    "                    'rerankingConfig':\n",
+    "                    {\n",
+    "                        'rerankerId': 272725718,\n",
+    "                        'mmrConfig': {\n",
+    "                            'diversityBias': 0.3\n",
+    "                        }\n",
+    "                    },\n",
+    "                    'summary': [\n",
+    "                        {\n",
+    "                            'responseLang': 'eng',\n",
+    "                            'maxSummarizedResults': 5,\n",
+    "                            'summarizerPromptName': self.prompt_name,\n",
+    "                            'debug': True\n",
+    "                        }\n",
+    "                    ]\n",
+    "                } \n",
+    "            ]\n",
+    "        }\n",
+    "        if self.prompt_text:\n",
+    "            body['query'][0]['summary'][0]['promptText'] = self.prompt_text\n",
+    "        return body\n",
+    "\n",
+    "    def get_headers(self):\n",
+    "        return {\n",
+    "            \"Content-Type\": \"application/json\",\n",
+    "            \"Accept\": \"application/json\",\n",
+    "            \"customer-id\": self.customer_id,\n",
+    "            \"x-api-key\": self.api_key,\n",
+    "            \"grpc-timeout\": \"60S\"\n",
+    "        }\n",
+    "\n",
+    "    def submit_query(self, query_str: str):\n",
+    "\n",
+    "        endpoint = f\"https://api.vectara.io/v1/query\"\n",
+    "        body = self.get_body(query_str)\n",
+    "\n",
+    "        response = requests.post(endpoint, data=json.dumps(body), verify=True, headers=self.get_headers())    \n",
+    "        if response.status_code != 200:\n",
```

Comment:
Unfortunately, this isn't the only way queries can respond with a failure.  You can get a 200 OK but have a failure embedded in the JSON.  We should check the response status after line 192 as in https://github.com/vectara/getting-started/blob/main/language-examples/python/vectara-rest/rest_query.py#L54